src/core/model/Document.cpp

```diff
@@ -136,6 +136,26 @@ auto Document::createSaveFoldername(const fs::path& lastSavePath) const -> fs::p
     return lastSavePath;
 }
 
+
+static std::string preprocessFormatString(const std::string& formatStr) {
+    std::string processed = formatStr;
```

Comment:
Since the string is copied anyway it would make more sense to pass it by value (and continue to work with `formatStr`)